streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -17,70 +17,158 @@
 package org.apache.kafka.streams.kstream.internals.suppress;
 
 import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.errors.StreamsException;
 import org.apache.kafka.streams.kstream.internals.Change;
+import org.apache.kafka.streams.kstream.internals.FullChangeSerde;
 import org.apache.kafka.streams.processor.Processor;
 import org.apache.kafka.streams.processor.ProcessorContext;
 import org.apache.kafka.streams.processor.internals.InternalProcessorContext;
+import org.apache.kafka.streams.processor.internals.ProcessorRecordContext;
 
-import java.time.Duration;
+import java.util.Iterator;
 
 import static java.util.Objects.requireNonNull;
 
 public class KTableSuppressProcessor<K, V> implements Processor<K, Change<V>> {
-    private final SuppressedImpl<K> suppress;
+    private final long maxRecords;
+    private final long maxBytes;
+    private final long suppressDurationMillis;
+    private final TimeOrderedKeyValueBuffer<Bytes, ContextualRecord<byte[]>> buffer;
+    private final TimeDefinition<K> timeDefinition;
+    private final BufferFullStrategy bufferFullStrategy;
+    private final boolean shouldSuppressTombstones;
     private InternalProcessorContext internalProcessorContext;
 
-    private final Serde<K> keySerde;
-    private final Serde<Change<V>> valueSerde;
+    private Serde<K> keySerde;
+    private Serde<Change<V>> valueSerde;
 
     public KTableSuppressProcessor(final SuppressedImpl<K> suppress,
                                    final Serde<K> keySerde,
                                    final Serde<Change<V>> valueSerde) {
-        this.suppress = requireNonNull(suppress);
+        requireNonNull(suppress);
         this.keySerde = keySerde;
         this.valueSerde = valueSerde;
+        maxRecords = suppress.getBufferConfig().maxRecords();
+        maxBytes = suppress.getBufferConfig().maxBytes();
+        suppressDurationMillis = suppress.getTimeToWaitForMoreEvents().toMillis();
+        buffer = new InMemoryTimeOrderedKeyValueBuffer();
+        timeDefinition = suppress.getTimeDefinition();
+        bufferFullStrategy = suppress.getBufferConfig().bufferFullStrategy();
+        shouldSuppressTombstones = suppress.shouldSuppressTombstones();
     }
 
     @Override
     public void init(final ProcessorContext context) {
         internalProcessorContext = (InternalProcessorContext) context;
+        this.keySerde = keySerde == null ? castSerde(context.keySerde()) : keySerde;
+        this.valueSerde =
+            valueSerde == null ? new FullChangeSerde<>(castSerde(context.valueSerde())) : valueSerde;
```

Comment:
Do we ever expect the passed in not-null valueSerde is a `FullChangeSerde` already? If not we should wrap it with `FullChangeSerde` still.

Reply:
This would mean that they have configured the `default.value.serde` as a FullChangeSerde, which is in the `internals` package.

Nevertheless, it doesn't hurt to guard it. Will do.

Reply:
Yeah I point is that is seems "impossible" that the passed in serde will be a `FullChangeSerde` but just the inner serde used for `FullChangeSerde`, so we should always wrap (either the default one from config, or the inherited one) with the `FullChangeSerde`, right?

Reply:
oh, I gotcha. The type of valueSerde is already a FullChangeSerde. In the case of an inherited serde, it gets wrapped in the constructor. The types ensure that the constructor arg is not already a FullChangeSerde.